File: scripts/calc.lua
-- input is a int provided by the host
print("Lua TID:", tid)
print("Sum to:", sum_to)
if (sum_to == 13) then error("unlucky number!") end
local accum = 0
for i = 0, sum_to do
    accum = accum + i
end
send(accum)

File: scripts/calc.dart
class Calc {
  int sum(List<Object> args) {
    var sumTo = args[0];

    if (sumTo == 13) {
      print("unlucky number!");
      invalid();
    }

    int accum = 0;
    for (int i = 0; i <= sumTo; i += 1) {
      accum = accum + i;
    }
    // print("acc: $sumTo -> $accum");
    return accum;
  }

  // non-std Dart, no need for async
  void loop(List<Object> args) {
    // non-std Dart, cant have empty for(;;;)
    for (true; true; true) {
      int result = sum(args);
      send("$result");
      sleep(1000); // non-std Dart, no need for await
    }
  }
}
